lib/commons/color/get-foreground-color.js

```diff
@@ -63,6 +64,11 @@ function getForegroundColor(node, _, bgColor) {
     return null;
   }
 
+  if (fgColor.alpha === 0) {
+    const textShadowColors = getTextShadowColors(node, { minRatio: 0 });
```

Comment:
How is this effected by offset and blur? I would like to see more tests on that. If blur is 0, even a small offset will result in the shadow sitting next to the text, instead of blending with it. The greater the blur, the less that is the case, but then the effect on the text color is greatly reduced.

Reply:
If blur is 0, and there is an offset greater than zero, currently `getTextShadowColors` treats that as not being a shadow that contributes to the text color at all.

Reply:
We'll just need a test for that to make sure.

Reply:
Added here. https://github.com/dequelabs/axe-core/pull/3001/commits/d3bf005a3c238c9b7e10c344b4d901d36dc98d2f